src/main/java/com/easypost/http/Requestor.java

```diff
@@ -553,7 +562,16 @@ private static <T> T httpRequest(final RequestMethod method, final String url, f
                     break;
             }
         }
+        Instant requestTimestamp = Instant.now();
+        UUID requestUuid = UUID.randomUUID();
+        Map<String, String> headers = new HashMap<String, String>();
+        headers = generateHeaders(client.getApiKey());
```

Comment:
What is happening on this line?

Reply:
It's generating the header for the request, the function requires api key because it's part of the request header

Reply:
nit: I think this stuff should be kicked further downstream, but I won't block on it. It's mostly just the header stuff that's not DRY here.